### PR TITLE
Add segmask support for RandomResizedCrop

### PR DESCRIPTION
**Pull request by ianstenbit**

### CODE REVIEW

**Comment by LukeWood** [comment]
@ianstenbit we will be changing the API for this in https://github.com/keras-team/keras-cv/pull/738

**Comment by LukeWood** [comment]
Looks like this PR will be (unchanged though)

**Comment by ianstenbit** [comment]
> Looks like this PR will be (unchanged though)

Ack -- Want to wait to merge this until after that PR?

**Review by LukeWood — APPROVED** [review]

File: keras_cv/layers/preprocessing/random_resized_crop.py [hunk 1]
```diff
@@ -194,6 +187,9 @@ def _check_class_arguments(
     def augment_target(self, augment_target, **kwargs):
         return augment_target
 
+    def augment_segmentation_mask(self, segmentation_mask, transformation, **kwargs):
+        return self._crop_and_resize(segmentation_mask, transformation)
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/random_resized_crop.py:191]: Will this have the same issue for mixing the pixel value when resize? eg if the [[0, 2], [0, 2]] is resize to [1, 1,]?

**Review by qlzh727 — CHANGES_REQUESTED** [review]

File: keras_cv/layers/preprocessing/random_resized_crop.py (hunk 1 above)
Comment (ianstenbit) [reply keras_cv/layers/preprocessing/random_resized_crop.py:191]: Yes -- I overlooked the interpolation issue for this layer. I will update this and send for re-review later today.

Comment (ianstenbit) [reply keras_cv/layers/preprocessing/random_resized_crop.py]: I've updated this implementation to use the build-in nearest-neighbor approach for interpolation, which handles the discrete class issue and avoids bad interpolation + rounding.

File: keras_cv/layers/preprocessing/random_resized_crop_test.py [hunk 2]
```diff
@@ -123,3 +123,28 @@ def test_crop_area_factor_errors(self, crop_area_factor):
...
+            target_size=self.target_size,
+            aspect_ratio_factor=(3 / 4, 4 / 3),
+            crop_area_factor=(0.8, 1.0),
+            seed=self.seed,
+        )
+
+        input_mask_resized = tf.image.resize(mask, self.target_size)
+
+        output = layer(input, training=True)
+        self.assertNotAllClose(output["segmentation_masks"], input_mask_resized)
```
Comment (qlzh727) [inline keras_cv/layers/preprocessing/random_resized_crop_test.py]: Can we do more strict assertion, eg make sure we don't mix the class of the mask?

**Review by qlzh727 — CHANGES_REQUESTED** [review]

File: keras_cv/layers/preprocessing/random_resized_crop_test.py (hunk 2 above)
Comment (ianstenbit) [reply keras_cv/layers/preprocessing/random_resized_crop_test.py]: I think this comment is based on an old diff -- the most recent version of this code does verify that we only output classes `0` and `7` using `self.assertAllInSet(output["segmentation_masks"], [0, 7])`

**Review by qlzh727 — APPROVED** [review]

**Pull request merged**